ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-security.j2

```diff
@@ -129,6 +129,45 @@
 {%         endfor %}
 {%     endif %}
+{%     if management_security.shared_secret_profiles is arista.avd.defined %}
+
+### Secret Profiles
+
+| Profile Name | Secret Name | Secret Password | Receive Lifetime | Transmit Lifetime |
+| ------------ | ----------- | --------------- | ---------------- | ----------------- |
+{%         for profile in management_security.shared_secret_profiles | arista.avd.natural_sort("profile_name") %}
+{%             if profile.secrets is arista.avd.defined %}
+{%                 set temp = [] %}
+{%                 for secret in profile.secrets | arista.avd.natural_sort("name") %}
+{%                     do temp.append(secret) %}
+{%                 endfor %}
+{%                 for secret in temp %}
+{%                     if temp.index(secret) == 0 and secret.lifetime.infinite is arista.avd.defined(true) %}
+| {{ profile.profile_name }} | {{ secret.name }} | {{ secret.secret | arista.avd.hide_passwords(hide_passwords) }} | infinite | infinite |
+{%                     elif temp.index(secret) == 0 and secret.receive_lifetime is arista.avd.defined and secret.transmit_lifetime is arista.avd.defined %}
+{%                         if secret.receive_lifetime.infinite is arista.avd.defined(true) and secret.transmit_lifetime.start_date_time is arista.avd.defined and secret.transmit_lifetime.end_date_time is arista.avd.defined %}
+| {{ profile.profile_name }} | {{ secret.name }} | {{ secret.secret | arista.avd.hide_passwords(hide_passwords) }} | infinite | {{ secret.transmit_lifetime.start_date_time }} - {{ secret.transmit_lifetime.end_date_time }} |
+{%                         elif secret.transmit_lifetime.infinite is arista.avd.defined(true) and secret.receive_lifetime.start_date_time is arista.avd.defined and secret.receive_lifetime.end_date_time is arista.avd.defined %}
+| {{ profile.profile_name }} | {{ secret.name }} | {{ secret.secret | arista.avd.hide_passwords(hide_passwords) }} | {{ secret.receive_lifetime.start_date_time }} - {{ secret.receive_lifetime.end_date_time }} | infinite |
+{%                         elif secret.receive_lifetime.start_date_time is arista.avd.defined and secret.receive_lifetime.end_date_time is arista.avd.defined and secret.transmit_lifetime.start_date_time is arista.avd.defined and secret.transmit_lifetime.end_date_time is arista.avd.defined %}
+| {{ profile.profile_name }} | {{ secret.name }} | {{ secret.secret | arista.avd.hide_passwords(hide_passwords) }} | {{ secret.receive_lifetime.start_date_time }} - {{ secret.receive_lifetime.end_date_time }} | {{ secret.transmit_lifetime.start_date_time }} - {{ secret.transmit_lifetime.end_date_time }} |
+{%                         endif %}
+{%                     endif %}
+{%                     if temp.index(secret) != 0 and secret.lifetime.infinite is arista.avd.defined(true) %}
```

Comment:
Please read up on Jinja2 loop.index to avoid this.

Reply:
Create a table for profile. So removed indexing.